kbase-extension/static/kbase/js/common/cellComponents/filePathWidget.js

```diff
@@ -47,190 +37,202 @@ define([
             events = Events.make();
 
         bus = runtime.bus().makeChannelBus({
-            description: 'A file path widget'
+            description: 'A file path widget',
         });
 
-        function makeFieldWidget(appSpec, parameterSpec, value, closeParameters) {
-            return paramResolver.loadInputControl(parameterSpec)
-                .then(function (inputWidget) {
-                    let fieldWidget = FieldWidget.make({
-                        inputControlFactory: inputWidget,
-                        showHint: true,
-                        useRowHighight: true,
-                        initialValue: value,
-                        appSpec: appSpec,
-                        parameterSpec: parameterSpec,
-                        workspaceId: workspaceId,
-                        referenceType: 'name',
-                        paramsChannelName: paramsBus.channelName,
-                        closeParameters: closeParameters
-                    });
-
-                    // Forward all changed parameters to the controller. That is our main job!
-                    fieldWidget.bus.on('changed', function (message) {
-                        paramsBus.send({
-                            parameter: parameterSpec.id,
-                            newValue: message.newValue,
-                            isError: message.isError
-                        }, {
-                            key: {
-                                type: 'parameter-changed',
-                                parameter: parameterSpec.id
-                            }
-                        });
+        function makeFieldWidget(inputWidget, appSpec, parameterSpec, value, closeParameters) {
```

Comment:
moved paramResolver call to line 386, to make this a synchronous function and easier to manage promise chain. the rest of this method is the same (aside from prettier edits)